starlette/middleware/cors.py

```diff
@@ -16,13 +17,18 @@ def __init__(
         allow_methods: typing.Sequence[str] = ("GET",),
         allow_headers: typing.Sequence[str] = (),
         allow_credentials: bool = False,
+        allow_origin_regex: str = None,
         expose_headers: typing.Sequence[str] = (),
         max_age: int = 600,
     ):
 
         if "*" in allow_methods:
             allow_methods = ALL_METHODS
 
+        if allow_origin_regex is not None:
+            regex = re.compile(allow_origin_regex)
+            allow_origin_regex = regex
```

Comment:
We're changing the type of this variable at this point, so I'd expect `mypy` will complain about this. I'll have a think about just tweaking how we set that.